activerecord/lib/active_record/persistence.rb

```diff
@@ -416,6 +416,8 @@ def toggle!(attribute)
     #   end
     #
     def reload(options = nil)
+      self.class.connection.clear_query_cache
```

Comment:
I think this should use `uncached` https://github.com/rails/rails/blob/master/activerecord/lib/active_record/connection_adapters/abstract/query_cache.rb#L47 block instead. So we dont clear the entire cache when calling reload.